### PR TITLE
[js/webgpu] Fix the transpose error when dims > 4D

### PR DESCRIPTION
**Pull request by qjia7**
### Description

Currently, the uniform support has bugs when dims rank is larger than 4. See https://github.com/microsoft/onnxruntime/issues/17860 item 1.
So this PR only enables shapes uniforms when shape rank is <= 4 for transpose. Otherwise, below compilation errors are thrown:
```
1 error(s) generated while compiling the shader:
:3:50 error: uniform storage requires that array elements are aligned to 16 bytes, but array element of type 'u32' has a stride of 4 bytes. Consider using a vector or struct as the element type instead.
      struct Uniforms { output_size:u32, a_shape:array<u32, 5>, a_strides:array<u32, 5>, output_shape:array<u32, 5>, output_strides:array<u32, 5> };
                                                 ^^^^^^^^^^^^^

:3:7 note: see layout of struct:
/*            align(4) size(84) */ struct Uniforms {
/* offset( 0) align(4) size( 4) */   output_size : u32;
/* offset( 4) align(4) size(20) */   a_shape : array<u32, 5>;
/* offset(24) align(4) size(20) */   a_strides : array<u32, 5>;
/* offset(44) align(4) size(20) */   output_shape : array<u32, 5>;
/* offset(64) align(4) size(20) */   output_strides : array<u32, 5>;
/*                              */ };
      struct Uniforms { output_size:u32, a_shape:array<u32, 5>, a_strides:array<u32, 5>, output_shape:array<u32, 5>, output_strides:array<u32, 5> };
      ^^^^^^

:4:42 note: 'Uniforms' used in address space 'uniform' here
      @group(0) @binding(2) var<uniform> uniforms: Uniforms;
                                         ^^^^^^^^
```

### CODE REVIEW

**Comment by qjia7** [comment]
@guschmue @satyajandhyala Please take a look, thanks.

FYI: sam-b-encoder meets this issue.

**Review by guschmue — APPROVED** [review]

**Comment by guschmue** [comment]
/mnt/vss/_work/1/s/js/web/lib/wasm/jsep/webgpu/ops/common.ts
  808:64  error  Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`  arrow-body-style
  810:2   error  Missing semicolon

File: js/web/lib/wasm/jsep/webgpu/ops/transpose.ts [hunk 1]
```diff
@@ -35,13 +35,18 @@ const permFunctionBody = (perm: number[], rank: number, input: IndicesHelper, ou
   return reverseFunc.join('\n');
 };
 
-export const createTransposeProgramInfo =
-    (inputDataType: number, inputRank: number, permAttr: number[]): ProgramInfo => {
-      const perm = getAdjustedPerm(inputRank, permAttr);
-      const output = outputVariable('output', inputDataType, (permAttr && permAttr.length) || inputRank);
-      const input = inputVariable('a', inputDataType, inputRank);
+export const createTransposeProgramInfo = (inputTensor: TensorView, permAttr: number[]): ProgramInfo => {
```
Comment (satyajandhyala) [inline js/web/lib/wasm/jsep/webgpu/ops/transpose.ts:38]: @qjia7  This change requires changes in other kernels using Transpose. Once this change is checked-in other kernels may be fixed. Did sam-b-encode work with this change?

Comment (qjia7) [reply js/web/lib/wasm/jsep/webgpu/ops/transpose.ts:38]: Great catching! I only run transpose tests and forgot to run others. Now everything is fixed. Thanks.

**Review by satyajandhyala — DISMISSED** [review]

**Review by satyajandhyala — APPROVED** [review]

**Comment by guschmue** [comment]
lint still nagging:
/mnt/vss/_work/1/s/js/web/lib/wasm/jsep/webgpu/ops/common.ts
  808:64  error  Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`  arrow-body-style

**Comment by qjia7** [comment]
> lint still nagging: /mnt/vss/_work/1/s/js/web/lib/wasm/jsep/webgpu/ops/common.ts 808:64 error Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>` arrow-body-style

@guschmue Currently, `npm run lint` is broken locally. So I only change the places you pointed out. Hope the lint can pass this time.
BTW, how can I find the errors you pasted?

**Comment by guschmue** [comment]
just tried the local 'npm run lint'. Yes, sorry, you are right, reports 'out of memory'. Let me see if I can fix this.

To see the details from the CI pipeline:
below in the CI results click on 'ONNX Runtime Web CI Pipeline' in case it failed, click on details, click on 'View more details on Azure Pipelines' and from there follow steps marked red.

**Comment by guschmue** [comment]
for eslint: on my windows box with nodejs v18.16.0 it is fine but fails on my linux box that runs nodejs v16.20.0. My linux box runs 16.04 and I can't update but the CI is on nodejs v18

**Comment by satyajandhyala** [comment]
> for eslint: on my windows box with nodejs v18.16.0 it is fine but fails on my linux box that runs nodejs v16.20.0. My linux box runs 16.04 and I can't update but the CI is on nodejs v18

`npm run lint` in onnxruntime\js folder failed with the following error on Thursday on Windows.

> lint
> eslint . --ext .ts --ext .tsx
 
 
<--- Last few GCs --->
 
[25420:000002255DE36030]   248472 ms: Scavenge 4032.8 (4118.1) -> 4031.6 (4118.6) MB, 6.2 / 0.0 ms  (average mu = 0.689, current mu = 0.442) allocation failure;
[25420:000002255DE36030]   250118 ms: Mark-sweep 4043.2 (4128.1) -> 4040.3 (4144.4) MB, 1611.2 / 0.0 ms  (average mu = 0.383, current mu = 0.056) allocation failure; scavenge might not succeed
 
 
<--- JS stacktrace --->
 
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
1: 00007FF69D06D51F node_api_throw_syntax_error+175743
2: 00007FF69CFF2E66 v8::internal::wasm::WasmCode::safepoint_table_offset+59654
3: 00007FF69CFF4B72 v8::internal::wasm::WasmCode::safepoint_table_offset+67090
4: 00007FF69DA9AAB4 v8::Isolate::ReportExternalAllocationLimitReached+116
5: 00007FF69DA85E12 v8::Isolate::Exit+674
6: 00007FF69D907C6C v8::internal::EmbedderStackStateScope::ExplicitScopeForTesting+124
7: 00007FF69D904E8B v8::internal::Heap::CollectGarbage+3963
8: 00007FF69D91B0C3 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath+2099
9: 00007FF69D91B96D v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath+93
10: 00007FF69D92B1A3 v8::internal::Factory::NewFillerObject+851
11: 00007FF69D61C285 v8::internal::DateCache::Weekday+1349
12: 00007FF69DB38051 v8::internal::SetupIsolateDelegate::SetupHeap+558193
13: 00007FF61F2F2D2D

**Pull request merged**